packages/twenty-server/src/engine/core-modules/guard-redirect/services/guard-redirect.service.ts

```diff
@@ -33,14 +38,20 @@ export class GuardRedirectService {
   }
 
   getSubdomainAndCustomDomainFromWorkspace(
-    workspace?: Pick<Workspace, 'subdomain' | 'customDomain'> | null,
+    workspace?: WorkspaceUrlBuilderComponentsType | null,
```

Comment:
if we need this type here, maybe this method should be extracted to the domain-manager too? to avoid logic duplication?